components/Navbar.jsx

```diff
@@ -1,14 +1,26 @@
-import Link from 'next/link'
-import react from 'react'
+import Link from "next/link";
+import react, { useEffect, useState } from "react";
+import { login, logout, onUserStateChange } from "../pages/api/firebase";
 
 export default function Navbar() {
+  const [user, setUser] = useState();
+
+  useEffect(() => {
+    onUserStateChange(setUser);
+  }, []);
+
   return (
     <nav>
-      <Link href="/"><h1>Hipapa</h1></Link>
+      <Link href="/">
+        <h1>Hipapa</h1>
+      </Link>
       <p>Welcome, teacher!</p>
-      <p>logout</p>
+      {!user ? (
```

Comment:
very small small thing and you don't have to change. But I find it easier to do user instead of !user.

less thinking.